pkg/instance/handler.go

```diff
@@ -102,80 +116,61 @@ func (h Handler) Create(c *gin.Context) {
 		return
 	}
 
-	savedInstance, err := h.instanceService.Create(instance)
-	if err != nil {
-		_ = c.Error(err)
-		return
-	}
+	if request.SourceInstance != 0 {
+		sourceInstance, err := h.instanceService.FindWithDecryptedParametersById(request.SourceInstance)
+		if err != nil {
+			_ = c.Error(err)
+			return
+		}
 
-	c.JSON(http.StatusCreated, savedInstance)
-}
+		canWriteSource := handler.CanWriteInstance(userWithGroups, sourceInstance)
+		if !canWriteSource {
+			err := apperror.NewUnauthorized(fmt.Sprintf("write access to source instance (id: %d) denied", sourceInstance.ID))
+			_ = c.Error(err)
+			return
+		}
 
-// Restart instance
-// swagger:route POST /instances/{id}/restart restartInstance
-//
-// Restart instance
-//
-// Security:
-//  oauth2:
-//
-// responses:
-//   202:
-//   401: Error
-//   403: Error
-//   404: Error
-//   415: Error
-func (h Handler) Restart(c *gin.Context) {
-	idParam := c.Param("id")
-	id, err := strconv.ParseUint(idParam, 10, 64)
-	if err != nil {
-		_ = c.AbortWithError(http.StatusBadRequest, fmt.Errorf("failed to parse id: %s", err))
-		return
-	}
+		if sourceInstance.DeployLog == "" {
```

Comment:
Is that the only information we have to check whether an instance is deployed?

Reply:
Yes, if there's nothing in the log it's not deployed. However this was the first place I had to do this.

We could wrap the evaluation (sourceInstance.DeployLog == "") in a method if you think that's cleaner?

Reply:
That's fine. I thought we might have a timestamp.